src/spikeinterface/extractors/cellexplorersortingextractor.py

```diff
@@ -1,136 +1,163 @@
+        except NotImplementedError:
+            spikes_mat = h5py.File(name=spikes_cellinfo_path, mode="r")
+            read_as_hdf5 = True
+
         if sampling_frequency is None:
-            folder_path = spikes_matfile_path.parent
-            sorting_id = spikes_matfile_path.name.split(".")[0]
-            if session_info_matfile_path is None:
-                session_info_matfile_path = folder_path / f"{sorting_id}.sessionInfo.mat"
-            session_info_matfile_path = Path(session_info_matfile_path)
-            assert (
-                (session_info_matfile_path).is_file()
-            ), f"No {sorting_id}.sessionInfo.mat file found in the folder!" 
-
-            try:
-                session_info_mat = scipy.io.loadmat(file_name=str(session_info_matfile_path))
-                self.read_session_info_with_scipy = True
-            except NotImplementedError:
-                session_info_mat = hdf5storage.loadmat(file_name=str(session_info_matfile_path))
-                self.read_session_info_with_scipy = False
-            
-            assert session_info_mat["sessionInfo"]["rates"][0][0]["wideband"], (
-                "The sesssionInfo.mat file must contain "
-                "a 'sessionInfo' struct with field 'rates' containing field 'wideband' to extract the sampling frequency!"
-            )
-            if self.read_session_info_with_scipy:
-                sampling_frequency = float(
-                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0][0][0]
-                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
-            else:
-                sampling_frequency = float(
-                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0]
-                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+            # First try is for the new format of spikes.cellinfo.mat files where sampling rate is included
+            sampling_frequency = spikes_mat["spikes"].get("sr", None)
+            sampling_frequency = sampling_frequency[()] if sampling_frequency is not None else None 
+        
+        if sampling_frequency is None:
+            sampling_frequency = self._retrieve_sampling_frequency_from_session_info()
+        
+        sampling_frequency = float(sampling_frequency)
+        
+        unit_ids_available = "UID" in spikes_mat["spikes"].keys()
+        assert unit_ids_available, "The spikes.cellinfo.mat file must contain field 'UID'!"
 
-        try:
-            spikes_mat = scipy.io.loadmat(file_name=str(spikes_matfile_path))
-            self.read_spikes_info_with_scipy = True
-        except NotImplementedError: 
-            spikes_mat = hdf5storage.loadmat(file_name=str(spikes_matfile_path))
-            self.read_spikes_info_with_scipy = False
+        spike_times_available = "times" in spikes_mat["spikes"].keys()
```

Comment:
Are the access patterns as dictionaries actually equivalent for both backends? (scipy vs. h5py)

I had thought that was a bigger part of the previous design

Reply:
scipy with the keyword argument `simplify_cells` returns dictionaries. The h5py returns groups who have keys as well.